docs/data/material/customization/palette/palette.md

```diff
@@ -108,16 +108,37 @@ const theme = createTheme({
 });
 ```
 
-As in the example above, if the palette color contains custom colors using any of the
-"main", "light", "dark" or "contrastText" keys, these map as follows:
+For the default palette colors (the ones listed in the [Default values section](#default-values) above), only the `main` token is required. If the other tokens are not provided their value is calculated as follows:
 
-- If the "dark" and / or "light" keys are omitted, their value(s) will be calculated from "main",
-  according to the "tonalOffset" value.
-- If "contrastText" is omitted, its value will be calculated to contrast with "main",
-  according to the "contrastThreshold" value.
+- The `dark` and `light` tokens are calculated according to the `tonalOffset` value with the `main` color as a starting point.
+- The `contrastText` token is calculated according to the `contrastThreshold` value to contrast with the `main` color.
 
-Both the "tonalOffset" and "contrastThreshold" values may be customized as needed.
-The "tonalOffset" value can either be a number between 0 and 1, which will apply to both light and dark variants, or an object with light and dark variants specified by the following TypeScript type:
+For custom colors, all tokens (`main`, `light`, `dark` and `contrastText`) are required. **Their values won't be automatically calculated** like for the default palette colors. If you wish to calculate the tokens from the `main` color you can use the palette's `augmentColor` utility, you'll have to create the theme in two steps:
+
+```js
+import { createTheme } from '@mui/material/styles';
+
+const theme = createTheme({
+  // Theme customization here as usual, including tonalOffset and/or
+  // contrastThreshold as the augmentColor util relies on these
+});
+
+const themeWithCustomColor = createTheme(theme, {
+  // Custom colors created with augmentColor here
+  palette: {
+    customColor: theme.palette.augmentColor({
+      color: {
+        main: '#FF5733',
+      },
+      name: 'customColor',
+    }),
+  },
+});
+```
```

Comment:
```suggestion
**Values for custom palette colors won't be automatically calculated** like for the default palette colors.
To generate color tokens from a custom `main` color, create the theme in two steps and use the palette's `augmentColor` utility:

```js
import { createTheme } from '@mui/material/styles';

const theme = createTheme({
  // Theme customization here as usual, including tonalOffset and/or
  // contrastThreshold as the augmentColor util relies on these
});

const themeWithCustomColor = createTheme(theme, {
  // Custom colors created with augmentColor here
  palette: {
    customColor: theme.palette.augmentColor({
      color: {
        main: '#FF5733',
      },
      name: 'customColor',
    }),
  },
});
```

Alternatively you can provide all tokens (`main`, `light`, `dark` and `contrastText`) manually - `@mui/system` provides [a set of utilities](https://github.com/mui/material-ui/blob/master/packages/mui-system/src/colorManipulator.d.ts) for manipulating color values.

Reply:
I think not saying that creating the theme in two steps is something that "you _have_ to do" makes it seem less like a weird workaround 😅

Also while looking at this PR I found that @mui/system exports a bunch of color utilities that aren't documented in the docs. I figure it might be worth linking to here – for somebody who wants to go with "provide all the tokens myself", it might save them from having to shop for a another color manipulation library 😬

Reply:
I think linking to somewhere in the source is enough for now ~